### PR TITLE
hack/Containerfile: New file for local container builds

### PR DESCRIPTION
**Pull request by cgwalters**
This is a relatively efficient container-oriented build flow suitable for local iteration.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +1,6 @@
 .cosa
+_kola_temp
+docs
+ci
```
Comment (jmarrero) [inline .dockerignore:4]: I think this is breaking:
https://github.com/containers/bootc/blob/fad73f26a49eec58b3a150dd3579e9ed7d3a92d6/ci/Dockerfile.ci#L7

**Review by jmarrero — CHANGES_REQUESTED** [review]

File: .dockerignore (hunk 1 above)
Comment (cgwalters) [reply .dockerignore]: Ah right, thanks for debugging this!  It's not actually important so dropped

**Comment by cgwalters** [comment]
OK exciting...CI just fell over with 

```
error[E0308]: mismatched types
  --> /github/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.29/src/backend/linux_raw/net/netdevice.rs:23:78
   |
23 |     unsafe { ifreq.ifr_ifrn.ifrn_name[..if_name_bytes.len()].copy_from_slice(if_name_bytes) };
   |                                                              --------------- ^^^^^^^^^^^^^ expected `&[i8]`, found `&[u8]`
   |                                                              |
   |                                                              arguments to this method are incorrect
   |
   = note: expected reference `&[i8]`
              found reference `&[u8]`
note: method defined here
  --> /builddir/build/BUILD/rustc-1.75.0-src/library/core/src/slice/mod.rs:3661:12

error[E0604]: only `u8` can be cast as `char`, not `i8`
  --> /github/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.29/src/backend/linux_raw/net/netdevice.rs:47:22
   |
47 |             .map(|v| *v as char)
   |                      ^^^^^^^^^^ invalid cast
   |
help: try casting from `u8` instead
  --> /github/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.29/src/backend/linux_raw/net/netdevice.rs:47:22
   |
47 |             .map(|v| *v as char)
   |                      ^^^^^^^^^^

Some errors have detailed explanations: E0308, E0604.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `rustix` (lib) due to 2 previous errors
Error: Process completed with exit code 101.
```

which seems clearly unrelated to this PR...hmmm....I wonder if this is another obscure incremental compilation bug.

**Comment by cgwalters** [comment]
Ah, this was https://github.com/bytecodealliance/rustix/commit/f546573ca5e66c0361f819068e213ade95afa1f7 but we aren't yet checking in `Cargo.lock`...perhaps it is time to do so.
/retest

**Pull request merged**